components/blog-article.tpl

```diff
@@ -8,7 +8,9 @@
       {% endif %}
     </h1>
 
-    <time class="article__date" datetime="{{ article.created_at | date: '%Y-%m-%d' }}">{{ article.created_at | format_date: "long" }}</time>
+    {% assign article_year = article.created_at | format_date: "%Y" | to_num %}
+
+    <time class="article__date" datetime="{{ article.created_at | date: '%Y-%m-%d' }}">{% if article_year < current_year %}{{ article.created_at | format_date: "long" }}{% else %}{{ article.created_at | format_date: "long_without_year" }}{% endif %}</time>
```

Comment:
* Move the *if/else* block to */components/blog-article-variables* and assign the proper value to a variable. For example `article_created_at`. Use this variable in here and in */layouts/blog___news.tpl* aswell.

* Change the logic so it will display the article year if the publishing date is in the future year.